src/hooks.ts

```diff
@@ -75,13 +75,22 @@ export function useProgress(updateInterval?: number) {
   const [state, setState] = useState<ProgressState>({ position: 0, duration: 0, buffered: 0 })
   const playerState = usePlaybackState()
   const stateRef = useRef(state)
+  const isUnmountedRef = useRef(true)
```

Comment:
ah, quite nice! you have tested this to work?
do you have an example app where I can see this failing before the fix?

Reply:
Actually, we probably want a similar fix for all hooks.

Reply:
I merged this, and applied the fix for `usePlaybackState`

Reply:
> ah, quite nice! you have tested this to work?
> do you have an example app where I can see this failing before the fix?

This problem happened occasionally in my application before, and it can work normally after applying this fix

---

> Actually, we probably want a similar fix for all hooks. Do you feel up for making those changes @lyswhut?

Okay, I will take time to check other asynchronous hooks when I have time

Reply:
No need I've applied the fix now also to `usePlaybackState` here: https://github.com/DoubleSymmetry/react-native-track-player/commit/abd2670156bd77a07c0fb7d909a8137fe93f010d

Thanks for the PR! It'll be out soon with the next version.